### DISCUSSION TITLE
tracking issue: permissions

### DISCUSSION THREAD

**Discussion by cafkafk** [eza-community/eza#152] [Development]
# On organization roles
> I created a request to join maintainers so I can move it there.

I looked into permissions for doing this, and I can only give "[repository roles](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization)" via teams on a repository level, not organizationally.

There are only two types of organizational roles:
![2023-08-21_15-16](https://github.com/eza-community/eza/assets/89321978/43449d1f-1135-42a4-8031-303f79f6be2a)

There is a feature to add more, [but it's paid](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/managing-custom-repository-roles-for-an-organization) :(

The best I can do in the free tier seems to be enabling all organization members to create public repositories, which I have just done.

# On team roles
I'm gonna think about the best way to structure repository permissions and then redo these very soon!

I think it makes sense to give out more permissions to you and @PThorpe92, so you can triage and do stuff like assign labels, or assigning yourself to issues etc, which I just realized today you couldn't!

Tell me if there is something you can't do that you should be able to do.

_Originally posted by @cafkafk in https://github.com/eza-community/eza/issues/150#issuecomment-1686326866_

**Comment by PThorpe92** [comment]
Definitely being able to assign / triage / tag issues would be nice. Also being able to review PR's, I don't care about merge/write privileges, I wouldn't ever use it unless the PR had already been signed off by everyone anyway so there's definitely no point. I'm sure it's hard to get more specific privileges assigned to people without paying for a premium account.

For what it's worth, it probably wouldn't hurt to give @gierens full maintainer privileges, for the "just-in-case" scenario, and because he is maintaining several of the dependencies + the deb package for us anyway. :+1: and he seems like a good guy.

**Reply by cafkafk** [reply]
> For what it's worth, it probably wouldn't hurt to give @gierens full maintainer privileges, for the "just-in-case" scenario, and because he is maintaining several of the dependencies + the deb package for us anyway. 👍 and he seems like a good guy.

I've added him to https://github.com/orgs/eza-community/teams/eza-maintainers and made the teams permissions write. I have to read more about perms before I do more, but this isn't final.

But yes, the just in case is very much something I try to keep in mind.

**Comment by PThorpe92** [comment]
@cafkafk I have some technical questions about the codebase from trying to specify the column order, I've been lost trying to figure some of this out, You or @gierens seem to be the most familiar with it. Especially since I would like to continue to help contribute/maintain, I'd really like to better understand some of this.

It would be cool to have something like a [gitter](https://gitter.im/) channel for this project, but I know there's only a few of us and I think we are all on different time zones. 

But idk what is your preferred method of communication? It's hard to get general dynamic questions answered just via posts/issues. Plus there's not really an appropriate place for them.
But either the email and linkedin on my profile are :+1: if you are comfortable reaching out, or if you'd rather something else let me know (anything except discord, which unfortunately isn't an option for me)

**Reply by cafkafk** [reply]
I created a room at #eza:gitter.im, I've been meaning to do this for a while just needed a push to do it :p
